src/decompose_mesh/module_database.f90

```diff
@@ -0,0 +1,457 @@
+
+    ! write MOHO 
+    ! optional moho
+    loc_nspec2D_moho = 0
+    do i=1,nspec2D_moho
+       if (ipart(ibelm_moho(i)) == myrank+1) then
+          loc_nspec2D_moho = loc_nspec2D_moho + 1
+       endif
+    enddo
+    ! checks if anything to do
+    if (loc_nspec2D_moho > 0) then
+
+       ! format: #surface_id, #number of elements
+       write(IIN_database) 7, loc_nspec2D_moho
+       
+       ! outputs element index and element node indices
+       ! note: assumes that element indices in ibelm_* arrays are in the range from 1 to nspec
+       !          (this is assigned by CUBIT, if this changes the following indexing must be changed as well)
+       !          while glob2loc_elmnts(.) is shifted from 0 to nspec-1  thus
+       !          we need to have the arg of glob2loc_elmnts start at 0 ==> glob2loc_nodes(ibelm_** -1)
+       
+       ! optional moho
+       do i=1,nspec2D_moho
+          if (ipart(ibelm_moho(i)) == myrank+1) then
+             do inode = 1,NGNOD2D
+                node_loc(inode) = glob2loc_nodes(nodes_ibelm_moho(inode,i))
+             enddo
+             write(IIN_database) glob2loc_elmnt(ibelm_moho(i)), node_loc(1:NGNOD2D)
+          endif
+       enddo
+    end if
+    close(IIN_database)
+    
+  end subroutine write_database
+
+!----------------------------------
+!
+!  count elements in boudary
+!
+!----------------------------------
+
+  subroutine count_my_boundary(myrank, ipart, ibelm, nspec2D, nE, iflag)
+
+    implicit none
+    integer,                                   intent(in) :: myrank, iflag
+    integer,                                   intent(in) :: nspec2D, nE
+    integer,   dimension(nE),                  intent(in) :: ipart
+    integer,   dimension(nspec2D),             intent(in) :: ibelm
+
+    integer ::  i, nspec2D_loc, iE
+    
+    nspec2D_loc = 0
+    do i=1, nspec2D
+       iE=ibelm(i)
+       if (ipart(iE) == myrank +1) nspec2D_loc = nspec2D_loc + 1
+    end do
+    write(IIN_database) iflag, nspec2D_loc
+
+  end subroutine count_my_boundary
+
+!----------------------------------
+!
+!  write elements in boudary
+!
+!----------------------------------
+
+  subroutine  write_my_boundary(myrank, ipart, ibelm, nodes_ibelm, nspec2D, nE)
+    implicit none
+
+    integer,                                   intent(in) :: myrank, nspec2D, nE
+    integer,   dimension(nE),                  intent(in) :: ipart
+    integer,   dimension(nspec2D),             intent(in) :: ibelm
+    integer,   dimension(NGNOD2D,nspec2D),     intent(in) :: nodes_ibelm
+
+    integer :: i,  iE, iE_loc, inode
+   
+    do i=1, nspec2D
+       iE=ibelm(i)
+       iE_loc=glob2loc_elmnt(iE)
+       do inode = 1, NGNOD2D
+          node_loc(inode) = glob2loc_nodes(nodes_ibelm(inode, i))
+       end do
+       if (ipart(iE) == myrank +1) write(IIN_database) iE_loc, node_loc(1:NGNOD2D) 
+    end do
+    
+  end subroutine write_my_boundary
+
+!----------------------------------
+!
+!  distributed adjancy table 
```

Comment:
Fixed.

On 02/10/2016 07:29 PM, Elliott Sales de Andrade wrote:
> In src/decompose_mesh/module_database.f90
> <https://github.com/geodynamics/specfem3d/pull/694#discussion_r52501360>:
>
>> +    integer :: i,  iE, iE_loc, inode
>> +
>> +    do i=1, nspec2D
>> +       iE=ibelm(i)
>> +       iE_loc=glob2loc_elmnt(iE)
>> +       do inode = 1, NGNOD2D
>> +          node_loc(inode) = glob2loc_nodes(nodes_ibelm(inode, i))
>> +       end do
>> +       if (ipart(iE) == myrank +1) write(IIN_database) iE_loc, node_loc(1:NGNOD2D)
>> +    end do
>> +
>> +  end subroutine write_my_boundary
>> +
>> +!----------------------------------
>> +!
>> +!  distributed adjancy table
>
> adjancy -> adjacency
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/specfem3d/pull/694/files#r52501360>.
>

-- 
Dimitri Komatitsch
CNRS Research Director (DR CNRS), Laboratory of Mechanics and Acoustics,
UPR 7051, Marseille, France    http://komatitsch.free.fr